### ISSUE TITLE
As a submitter and UO I need to be able to add or remove co-is after the proposal has been submitted 

### ISSUE THREAD

**Issue by WhittyPM** [UserOfficeProject/stfc-user-office-project#284]
UAT output

**Comment by IVlaD17** [comment]
After investigating this issue more, it appears that the bug can only be found when dealing with submitted proposals from the perspective of a User Officer and PIs can easily add Co-Investigators.

**Comment by IVlaD17** [comment]
After looking into this even more, there are a couple of solutions that are present, but it's important to understand the whole situation.

Therefore, for context, the website uses two different tables to add users to a proposal after it was submitted depending on whether the user trying to do this is a user officer or an user:
- If a user tries to add another co-investigator after the proposal was submitted, the app uses the `ProposalsPeopleTable` which has a search bar that searches the database for users based on their email address
- If a user officer tries to add another co-investigator after the proposal was submitted, the app uses the `PeopleTable` which is meant to display all users in the database (this is functionality that works for the ESS) and it also has a search bar through which you can filter the users.

Because we have a different database setup/structure compared to the ESS, this second table doesn't work for us, therefore, we need to change the way we add users as user officers and we've got multiple solutions for this, each with advantages and disadvantages:
- We can replace the `PeopleTable` with `ProposalsPeopleTable` for User Officers.
  - Main advantage of this is that the change is fairly straight forward: it's a simple `if` statement change and then replacing the tests
  - Main disadvantage of this is that the ESS might very much want to keep the `PeopleTable` in which case we'll have to implement a feature and that's going to impact the tests as well
- We can add the 'search bar' functionality from the `ProposalsPeopleTable` to the `PeopleTable`.
  - Main advantage of this is that the ESS would get to keep the core functionality of the `PeopleTable` while we'd also get the functionality that we need
  - Main disadvantage is that it might make the ESS user experience a bit more confusing as well as being somewhat more complicated to implement

**Comment by ACLay** [comment]
As I see it, `ProposalsPeopleTable` doesn't seem right for the User Officer view. It seems to be a UI component for getting recent collaborators, (or individual searchable users) which I don't think applies here. What exactly is the problem with the PeopleTable?

**Comment by IVlaD17** [comment]
> As I see it, `ProposalsPeopleTable` doesn't seem right for the User Officer view. It seems to be a UI component for getting recent collaborators, (or individual searchable users) which I don't think applies here. What exactly is the problem with the PeopleTable?

Because of the way our databases/APIs work, there's no way for us to get all the users for the `PeopleTable` to display correctly, therefore, it is completely bugged and there is no way we can fix that at the moment.

**Comment by WhittyPM** [comment]
@IVlaD17 @TickleThePanda  would it be worth having a team review of this? My thoughts are that it's not a HUGE issue do get it fixed now, but it is an expectation from the UO. The UO could go back to the Pi and ask them to add as they have that functionality (is that correct?). Just so we are clear, the UO can edit anything else, it's just they can't add a new collaborator onto the proposal?
Thanks Vlad for putting all the investigation work into this.

**Comment by IVlaD17** [comment]
> @IVlaD17 @TickleThePanda would it be worth having a team review of this? My thoughts are that it's not a HUGE issue do get it fixed now, but it is an expectation from the UO. The UO could go back to the Pi and ask them to add as they have that functionality (is that correct?). Just so we are clear, the UO can edit anything else, it's just they can't add a new collaborator onto the proposal? Thanks Vlad for putting all the investigation work into this.

Yeah. I'm hoping we could have that in this thread. I've also posted on teams but I can make people aware of this tomorrow during stand-up.
